Chessington.GameEngine/Pieces/King.cs

```diff
@@ -10,7 +10,28 @@ public King(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var initialRow = board.FindPiece(this).Row;
+            var initialCol = board.FindPiece(this).Col;
+            var availableMoves = new List<Square>();
+            
+            for (int i = -1; i <= 1; i++)
+            {
+                for (int j = -1; j <= 1; j++)
+                {
+                    if (CheckPosition(initialRow + i, initialCol + j))
+                    {
+                        availableMoves.Add(Square.At(initialRow + i, initialCol + j));
+                    }
+                }
+            }
+            
+            availableMoves.RemoveAll(x => x == Square.At(initialRow,initialCol));
+            
+            return availableMoves;
+        }
+        private bool CheckPosition(int row, int col)
```

Comment:
This method is exactly same on bishop and Rook. You can move it up to super class which class "Piece" and just have one copy of it and call it in all subclasses